lib/tools/adb-commands.js

```diff
@@ -707,8 +707,25 @@ methods.getDeviceProperty = async function (property) {
 };
 
 methods.setDeviceProperty = async function (prop, val) {
+  let apiLevel = await this.getApiLevel();
+  if (apiLevel >= 26 ) {
+    log.debug(`Running adb root, Android O needs adb to be rooted to setDeviceProperty`);
+    await this.root();
+  }
   log.debug(`Setting device property '${prop}' to '${val}'`);
-  await this.shell(['setprop', prop, val]);
+  let err;
+  try {
+    await this.shell(['setprop', prop, val]);
+  } catch (e) {
+    err = e;
+  }
+  if (apiLevel >= 26 ) {
+    log.debug(`Removing adb root for setDeviceProperty`);
+    await this.unroot();
+  }
+  if (_.isError(err)) {
```

Comment:
@imurchie it was me who asked for this change. IMHO positive conditions are easier to read than negative (it was `!._isUndefined(err)`). But yeah, `if (err)` is also very readable